src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,50 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      const index = Animal.alive.indexOf(this);
+
+      if (index > -1) {
+        Animal.alive.splice(index, 1);
+      }
+    }
+  }
```

Comment:
[CODE PERFOMANCE] - Avoid nesting loops (indexOf + splice). Instead, you can use the filter method to remove animals with health less than or equal to 0 from the Animal.alive array. This will improve the performance of your code.